src/client/pages/leaderboard/components/PokemonTile.jsx

```diff
@@ -1,75 +1,90 @@
-import React, { useState } from "react";
-import { Rating } from "react-simple-star-rating";
+import { useState } from 'react';
+import { Rating } from 'react-simple-star-rating';
+
 const PokemonTile = (props) => {
-  const { position, pokemons } = props;
-  const [rating, setRating] = useState(0); // initial rating value
+    const { topRatedPokemon, calcAverageRating } = props;
 
-  // Catch Rating value
-  const handleRating = (rate) => {
-    setRating(rate);
-    // other logic
-  };
-  return (
-    pokemons.length > 0 && (
-      <div className="pokemon-tile">
-        <img
-          className="top-3-pokemon-image"
-          src={pokemons[position].pictureUrl}
-          alt={pokemons[position].name}
-        />
-        <div className="pokemon-text-container">
-          <h3 className="pokemon-text-container-header">
-            {pokemons[position].name}
-          </h3>
-          <div className="stats">
-            <div className="stats-head">
-              <span>Hp:</span>
-              <span>Atk:</span>
-              <span>Def:</span>
-              <span>Spd:</span>
-              <span>Avg Rating: </span>
-            </div>
-            <div className="stats-values">
-              <span>{pokemons[position].baseHP}</span>
-              <span>{pokemons[position].baseAttack}</span>
-              <span> {pokemons[position].baseDefense}</span>
-              <span>{pokemons[position].speed}</span>
-              <span>4</span>
-            </div>
+    const [rating, setRating] = useState(0);
+
+    const handleRating = (rate) => setRating(rate);
```

Comment:
You already have this function in a parent component. Shouldn't we move it to a utils file?

Reply:
I tired moving state into the parent component, it broke the code, by applying the same rating to all Pokémon. I think this should stay as is.

Reply:
Saw the effects, agreed